pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -271,12 +293,61 @@ func (m *ManifestWorkController) applyOneManifest(
 	// TODO we should check the certain error.
 	// Use dynamic client when scheme cannot decode manifest or typed client cannot handle the object
 	if isDecodeError(result.Error) || isUnhandledError(result.Error) || isUnsupportedError(result.Error) {
-		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, unstructuredObj, gvr, recorder)
+		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, required, gvr, recorder)
 	}
 
 	return result
 }
 
+func (m *ManifestWorkController) serverSideApply(
+	ctx context.Context,
+	gvr schema.GroupVersionResource,
+	required *unstructured.Unstructured,
+	config *workapiv1.ServerSideApplyConfig,
+	recorder events.Recorder) (*unstructured.Unstructured, bool, error) {
+	force := false
+	fieldManager := "work-agent"
```

Comment:
nit: move it to public variables so that we can share the defaults from other places

Reply:
create a PR to put it in api repo https://github.com/open-cluster-management-io/api/pull/164